src/codegen/x86_64/arch_x86_64_isel.h

```diff
@@ -0,0 +1,13 @@
+#ifndef ARCH_X86_64_ISEL_H
+#define ARCH_X86_64_ISEL_H
+
+#include <codegen/codegen_forward.h>
+#include <codegen/machine_ir.h>
+
+/// Given Intercept IR, return a vector of machine instructions for x86_64 that represents the program.
+MIRInstructionVector select_instructions(CodegenContext *context);
+
+/// Given freshly-lowered MIR, return a vector of machine instructions for x86_64 that represents the program.
+MIRFunctionVector select_instructions2(MIRFunctionVector input);
```

Comment:
Er, why do two of these exist? Generally, it should probably always go   
AST -> IR (-> Optimiser) -> MIR -> ISel -> RA -> Emit.

Reply:
There are two parallel uses of the MIR right now. The whole "MIRValuex86_64` thing is what's used  by `select_instructions` and actually provides enough data to emit machine code from (or at least did). `select_instructions2` is the "dev branch" basically and is where I am actually working on things still. Both of them are used in parallel in different ways by the x86_64 backend right now, and eventually we'll stop using the first one and rename `2` to just `select_instructions`, but baby steps, ya know :P

Reply:
Ah yeah, I see how it is